### PR TITLE
feat(api): Include PII meta data in API context

### PR DESCRIPTION
**Pull request by jan-auer**
This PR exposes `_meta` generated during PII stripping and normalization to the UI. 

It introduces a new method `get_api_meta(self, meta, is_public=False)` on interfaces that should map the provided meta data to the same schema as `get_api_context` would return. The return value is then piped through `meta_with_chunks` which splits strings into chunks for rendering.

Note that we perform more normalization in event_manager and interfaces, so some _meta keys might still not be correct. Depending on how we move forward with the store refactor, this might be solved implicitly.

**This requires a bump to `semaphore:0.2.0` in getsentry**

### CODE REVIEW

File: src/sentry/api/serializers/models/event.py
Comment (jan-auer) [inline src/sentry/api/serializers/models/event.py]: **Important**: This was `.to_json()` and is now `.get_api_context()`. The UI expects `user.ip_address` and API context was dead code. The only use of `ipAddress` is in _groupTagValues.jsx_.

**Comment by mitsuhiko** [comment]
I think this is about as nice as it gets but it needs another pair of eyes to make sure this is sensible. This is basically the version of the design we agreed on a few months back. We also attempted another version with inline data but that is worse.

Note: the `ip_address` field here was apparently always named as the api export method was bypassed. This means external users already had to use `ip_address` as a field. We probably need to stick with this forever now.

File: src/sentry/event_manager.py
Comment (mitsuhiko) [inline src/sentry/event_manager.py]: I'm okay with this for now but I think we should probably leave a comment here why we do this and what we want to do going forward. There is a TODO item on the protocol iteration list for this logentry vs message thing.

File: src/sentry/interfaces/base.py
Comment (mitsuhiko) [inline src/sentry/interfaces/base.py]: can we change this to `if type(self) is not type(other)`? Isinstance checks should not be done here.

File: src/sentry/interfaces/message.py
Comment (mitsuhiko) [inline src/sentry/interfaces/message.py]: So this never worked? Do we want to do this then instead of just deleting the code?

**Review by mitsuhiko — CHANGES_REQUESTED** [review]

File: src/sentry/event_manager.py
Comment (jan-auer) [reply src/sentry/event_manager.py]: The comment is above that block, quite elaborate, actually.

File: src/sentry/interfaces/base.py
Comment (jan-auer) [reply src/sentry/interfaces/base.py]: The linter automatically changes this for me :(

File: src/sentry/interfaces/message.py
Comment (jan-auer) [reply src/sentry/interfaces/message.py]: Would love to kill that, actually! Could someone else please confirm, just to be safe?

File: src/sentry/interfaces/base.py
Comment (jan-auer) [reply src/sentry/interfaces/base.py]: No dice. The linter overrules this.

File: src/sentry/interfaces/message.py
Comment (bretthoerner) [reply src/sentry/interfaces/message.py]: Yeah, I don't see how this could have handled anything before so it seems like we can drop it.

File: src/sentry/interfaces/stacktrace.py
Comment (bretthoerner) [inline src/sentry/interfaces/stacktrace.py]: :heart_eyes: Thank you.

**Comment by bretthoerner** [comment]
I'll be honest, I read this all to start getting context but I am pretty unfamiliar with the semaphore/marshal stuff that's been going on and I'm catching up. It seems to make sense to me but I'm no expert here yet.

**Comment by jan-auer** [comment]
@mitsuhiko This contains minimal tests now. I propose cleaning this up (especially wrt the expected format of contents in _meta) once we start cleaning up the normalization code. Ideally, sentry code does not have to account for different representations of certain attributes if they **always** get normalized.

**Review by mitsuhiko — APPROVED** [review]
lgtm but when we merge this we must also raise the pin for semaphore in getsentry or this will break the build.

**Pull request merged**
